src/libstore/globals.cc

```diff
@@ -77,6 +77,23 @@ Settings::Settings()
     if (sslOverride != "")
         caFile = sslOverride;
 
+#ifdef __APPLE__
+    if(caFile.get().starts_with("keychain:")){
+        debug("reading %s",caFile.get());
+        auto caContents = runProgram("/usr/bin/security", false, {"find-certificate", "-a", "-p", caFile.get().substr(9)});
+        if (caContents.empty()){
+            warn("reading '%s' found no certificates",caFile.get());
+        }
+        auto caFilePath = settings.nixConfDir + "/ssl-cert-file.keychain";
+        auto caFilePathTmp = caFilePath + ".tmp";
```

Comment:
This looks still racy to me in case two processes attempt to write to the file. Using `mkstemp` with a prefix in the same directory better.

Reply:
Do we need it to be a single filename, or could we content-address it?